targets/TARGET_STM/sleep.c

```diff
@@ -40,17 +40,25 @@ extern void HAL_ResumeTick(void);
 
 void hal_sleep(void)
 {
+    // Disable IRQs
+    core_util_critical_section_enter();
```

Comment:
Okay, change since 5.4 identified.

"release" build profile has always slept, "debug" build profile has never slept.

The behaviour of the "develop" (ie default) build profile changed between in 5.4 and 5.5. In 5.4 it did not sleep - it was switched on NDEBUG. In 5.5 it does sleep - it is switched on MBED_DEBUG.

I've discussed the sleep a bit with @bulislaw, I think my previous comment is correct, so I'll rechick this and probably approve.